src/pip/_internal/utils/packaging.py

```diff
@@ -68,3 +75,30 @@ def is_pinned(specifier: SpecifierSet) -> bool:
             continue
         return True
     return False
+
+
+def filter_manylinux_tags(
+    glibc: Tuple[int, int], arch: str
+) -> Generator[str, None, None]:
+    for tag in filter(lambda t: t.startswith("manylinux"), platform_tags()):
+        tag_prefix, _, tag_suffix = tag.partition("_")
+        if tag_prefix in _LEGACY_MANYLINUX_MAP:
+            tag_glibc = _LEGACY_MANYLINUX_MAP[tag_prefix]
+            tag_arch = tag_suffix
+        else:
+            tag_glibc_major, tag_glibc_minor, tag_arch = tag_suffix.split("_", 2)
+            tag_glibc = (int(tag_glibc_major), int(tag_glibc_minor))
+
+        if arch == tag_arch and tag_glibc <= glibc:
+            yield tag
+
+
+def filter_musllinux_tags(
+    musl: Tuple[int, int], arch: str
+) -> Generator[str, None, None]:
+    for tag in filter(lambda t: t.startswith("musllinux"), platform_tags()):
+        *_, tag_suffix = tag.partition("_")
+        tag_musl_major, tag_musl_minor, tag_arch = tag_suffix.split("_", 2)
+        tag_musl = (int(tag_musl_major), int(tag_musl_minor))
+        if tag_arch == arch and tag_musl <= musl:
+            yield tag
```

Comment:
I think we can merge these two? A musllinux tag is never in `_LEGACY_MANYLINUX_MAP`, and if we replace `glibc` and `musl` with `libc`, the two implementations seem to match (if I didn’t miss anything).

Reply:
hmm, for glibc 
> (sys_major, sys_minor) >= (tag_major, tag_minor)

for musl
> A musllinux_1_1 wheel containing symbols only available in musl 1.2 or later.

So for glibc >= , for musl > ?

Reply:
It should be `>=` for both; or at least it should because I intentionally modelled the musllinux spec on PEP 600.

Reply:
Rewritten. How to mock libc this in functional tests?

Reply:
I don’t think we want to go so far to patching C libraries… Unit tests should be enough.